src/components/ArticleList.jsx

```diff
@@ -6,14 +6,17 @@ import { useParams } from "react-router-dom";
 const ArticleList = () => {
   let { topic } = useParams();
 
+  const [isLoading, setIsLoading] = useState(true);
+
   const [articles, setArticles] = useState([]);
 
   useEffect(() => {
     getArticles(topic).then(({ articles }) => {
       setArticles(articles);
+      setTimeout(() => setIsLoading(false), 2000);
```

Comment:
no need for a `setTimeout` here!

You can set loading to false as soon as it stops loading (i.e when the response comes back!)

At the moment we still get lag when switching topics